<?xml version="1.0"?>

<launch>
	<!--Launch move_base-->

	<arg name="use_simulation_topics" default="true" />
	<arg name="amcl" default="true" />

	<arg name="base_global_planner" default="navfn/NavfnROS"/>
	<arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
	<!--arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/-->

	<node pkg="move_base" type="move_base" respawn="false" name="NavigationNode" output="screen">
		
		<param name="base_global_planner" value="$(arg base_global_planner)"/>
		<param name="base_local_planner" value="$(arg base_local_planner)"/>

		<rosparam file="$(find human_aware_navigation)/config/planner.yaml" command="load"/>

		<!-- Observation sources located in costmap_common_params.yaml -->
		<rosparam file="$(find human_aware_navigation)/config/costmap_common_params_real_robot.yaml" command="load" ns="global_costmap"/>
		<rosparam file="$(find human_aware_navigation)/config/costmap_common_params_real_robot.yaml" command="load" ns="local_costmap"/>

		<!--rosparam file="$(find human_aware_navigation)/config/costmap_common_params_simulation.yaml" command="load" ns="global_costmap" if="$(arg use_simulation_topics)"/>
		<rosparam file="$(find human_aware_navigation)/config/costmap_common_params_simulation.yaml" command="load" ns="local_costmap" if="$(arg use_simulation_topics)"/-->

		<!-- Local Costmap -->
		<rosparam file="$(find human_aware_navigation)/config/costmap_local_params.yaml" command="load" ns="local_costmap" />
		<param name="local_costmap/width" value="10.0"/>
		<param name="local_costmap/height" value="10.0"/>

		<!-- static global costmap, static map provides size -->
		<rosparam file="$(find human_aware_navigation)/config/costmap_global_params_static.yaml" command="load" ns="global_costmap" if="$(arg amcl)"/>


		<!-- Remap some topics for a custom network -->
		<remap from="cmd_vel" to="cmd_vel_mux/input/navi" unless="$(arg use_simulation_topics)"/>
		<remap from="cmd_vel" to="husky_velocity_controller/cmd_vel" if="$(arg use_simulation_topics)"/>

		<!--remap from="odom" to="/odometry/filtered"/-->

	</node>
</launch>
